engine/execution/storehouse.go

```diff
@@ -0,0 +1,74 @@
+package execution
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// RegisterStore is the interface for register store
+// see implementation in engine/execution/storehouse/register_store.go
+type RegisterStore interface {
+	// GetRegister first try to get the register from InMemoryRegisterStore, then OnDiskRegisterStore
+	GetRegister(height uint64, blockID flow.Identifier, register flow.RegisterID) (flow.RegisterValue, error)
+
+	// SaveRegisters saves to InMemoryRegisterStore first, then trigger the same check as OnBlockFinalized
+	// Depend on InMemoryRegisterStore.SaveRegisters
+	SaveRegisters(header *flow.Header, registers []flow.RegisterEntry) error
+
+	// Depend on FinalizedReader's GetFinalizedBlockIDAtHeight
+	// Depend on ExecutedFinalizedWAL.Append
+	// Depend on OnDiskRegisterStore.SaveRegisters
+	// OnBlockFinalized trigger the check of whether a block at the next height becomes finalized and executed.
+	// the next height is the existing finalized and executed block's height + 1.
+	// If a block at next height becomes finalized and executed, then:
+	// 1. write the registers to write ahead logs
+	// 2. save the registers of the block to OnDiskRegisterStore
+	// 3. prune the height in InMemoryRegisterStore
+	OnBlockFinalized() error
+
+	// LastFinalizedAndExecutedHeight returns the height of the last finalized and executed block,
+	// which has been saved in OnDiskRegisterStore
+	LastFinalizedAndExecutedHeight() uint64
```

Comment:
Isn't a height that's finalized always executed as well? could this be: `LastFinalizedHeight` instead?

Reply:
Right, I still prefer to be explicit.